Patch to convert the cdbdump Utility from the public-domain-Package
„freecdb-0.75“ to a Utility to dump a binary Index

diff --git a/cdbdump.c b/cdbdump.c
index cbf2399..ef359dd 100644
--- a/cdbdump.c
+++ b/cdbdump.c
@@ -6,7 +6,7 @@
 #include "fmt.h"
 
 #define USAGE ""
-#define FATAL "cdbdump: fatal: "
+#define FATAL "cdbdumpindex: fatal: "
 
 #define BUFSIZE 1024
 
@@ -31,13 +31,12 @@ void put1(char *s, unsigned int l) {
     strerr_die2sys(111, FATAL, "unable to write output: ");
 }
 
-void copyb(uint32 len) {
+void skipb(uint32 len) {
   uint32 l;
 
   while (len) {
     l =len < BUFSIZE ? len : BUFSIZE;
     getb(buf, l);
-    put1(buf, l);
     len -=l;
   }
 }
@@ -65,6 +64,7 @@ int main(int argc, char **argv) {
 
   /* read keylen, datalen, key, data up to data size */
   while (pos < eodata) {
+    put1(&pos, 4);
     /* keylen */
     getb(buf, 4);
     pos +=4;
@@ -73,24 +73,15 @@ int main(int argc, char **argv) {
     getb(buf, 4);
     pos +=4;
     uint32_unpack(buf, &datalen);
-    /* format output keylen, datalen */
-    put1("+", 1);
-    put1(num, fmt_ulong(num, keylen));
-    put1(",", 1);
-    put1(num, fmt_ulong(num, datalen));
-    put1(":", 1);
-    /* format output key and data */
-    copyb(keylen);
+    /* ski key and data */
+    skipb(keylen);
     pos +=keylen;
-    put1("->", 2);
-    copyb(datalen);
+    skipb(datalen);
     pos +=datalen;
-    put1("\n", 1);
     /* next entry */
   }
 
   /* final empty line */
-  put1("\n", 1);
   if (buffer_flush(buffer_1) == -1)
     strerr_die2sys(111, FATAL, "unable to flush output: ");
   _exit(0);
